### PR TITLE
Adding helper function for getting Savefilepath

### PR DESCRIPTION
**Pull request by silverhadch**
Repeated logic implemnted in a clean and better way.  !!!! But please test the code i havent yet just wrote it on my Phone !!!!

### CODE REVIEW

File: plastic_tui/src/ui.rs [hunk 1]
```diff
@@ -537,6 +534,21 @@ impl Ui {
...
+                       if let Some(ref mut player) = self.audio_player {
+                let audio_buffer = process_audio(&audio_buffer, 1.0);
+                player.queue(&audio_buffer);
+            }
+
+            if let Some(remaining) = fps.remaining() {
+                thread::sleep(remaining);
+            }
+        }
+
```
Comment (Amjad50) [inline plastic_tui/src/ui.rs:551]: This got duplicated and is causing build errors

File: plastic_tui/src/ui.rs [hunk 2]
```diff
@@ -161,33 +161,30 @@ impl Ui {
...
-
-        let file = fs::File::open(&path).unwrap();
+    fn get_save_state_path(&self, slot: u8) -> Option<std::path::PathBuf> {
+        if self.nes.is_empty() {
+            return None;
+        }
 
-        self.nes.load_state(&file).unwrap();
+        let base_saved_states_dir = base_save_state_folder().ok()?;
+        let filename = self.nes.save_state_file_name(slot).ok()?;
```
Comment (Amjad50) [inline plastic_tui/src/ui.rs:183]: These are `Option`, so no need for `ok`

File: plastic_tui/src/ui.rs (hunk 1 above)
Comment (silverhadch) [reply plastic_tui/src/ui.rs:551]: Bruh how did that even happen. Lol

File: plastic_tui/src/ui.rs [hunk 3]
```diff
@@ -536,14 +534,5 @@ impl Ui {
...
         }
-
-        disable_raw_mode().unwrap();
-        execute!(
-            io::stdout(),
-            Show,
-            LeaveAlternateScreen,
-            PopKeyboardEnhancementFlags,
-        )
-        .unwrap();
```
Comment (Amjad50) [inline plastic_tui/src/ui.rs:547]: You removed this

File: plastic_tui/src/ui.rs [hunk 4]
```diff
@@ -161,32 +161,30 @@ impl Ui {
...
+        if let Some(path) = self.get_save_state_path(slot) {
+            let file = fs::File::open(&path).unwrap();
+            self.nes.load_state(&file).unwrap();
         }
+    }
+
+    fn get_save_state_path(&self, slot: u8) -> Option<std::path::PathBuf> {
+    if self.nes.is_empty() {
+        return None;
+    }
```
Comment (Amjad50) [inline plastic_tui/src/ui.rs:180]: formatting got messed here.

File: plastic_tui/src/ui.rs [hunk 5]
```diff
@@ -161,32 +161,30 @@ impl Ui {
...
+            return None;
         }
 
-        let base_saved_states_dir = base_save_state_folder().unwrap();
-        let filename = self.nes.save_state_file_name(slot).unwrap();
-        let path = base_saved_states_dir.join(&filename);
+    let base_saved_states_dir = base_save_state_folder()?;
+    let filename = self.nes.save_state_file_name(slot)?;
+    Some(base_saved_states_dir.join(filename))
+}
```
Comment (Amjad50) [inline plastic_tui/src/ui.rs:185]: You only fixed part of it, this still isn't formatted

**Comment by silverhadch** [comment]
For Linux a Test returns:
{"message":"Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 3316s."}

Is that something to worry about the commit or a general error?

**Comment by Amjad50** [comment]
> For Linux a Test returns: {"message":"Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 3316s."}
> 
> Is that something to worry about the commit or a general error?

you can ignore this, its with codecov, I'll fix it later.

**Comment by Amjad50** [comment]
Anyway thanks for this!

I'll merge it

**Pull request merged**